### ISSUE TITLE
MLIR Textual PassPipeline Error on VMFB load (Unet512 & VAE)

### ISSUE THREAD

**Issue by njsharpe** [nod-ai/SHARK#1902]
I've successfully recreated this issue on multiple, unrelated models. This issue does not seem to break model generation, and images are created as normal, but it may have an affect on image generation time (i.e. it/s), as can be seen by the generation logs below.

```
Found device AMD Radeon RX 7900 XTX. Using target triple rdna3-7900-windows.
Tuned models are currently not supported for this setting.
saving euler_scale_model_input_1_512_512_vulkan_fp16_torch_linalg.mlir to C:\Users\nelso\AppData\Local\Temp
No vmfb found. Compiling and saving to D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\euler_scale_model_input_1_512_512_vulkan_fp16.vmfb
Configuring for device:vulkan://00000000-5500-0000-0000-000000000000
Using target triple -iree-vulkan-target-triple=rdna3-7900-windows from command line args
Saved vmfb in D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\euler_scale_model_input_1_512_512_vulkan_fp16.vmfb.
WARNING: [Loader Message] Code 0 : windows_read_data_files_in_registry: Registry lookup failed to get layer manifest files.
Loading module D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\euler_scale_model_input_1_512_512_vulkan_fp16.vmfb...
        Compiling Vulkan shaders. This may take a few minutes.
saving euler_step_1_512_512_vulkan_fp16_torch_linalg.mlir to C:\Users\nelso\AppData\Local\Temp
No vmfb found. Compiling and saving to D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\euler_step_1_512_512_vulkan_fp16.vmfb
Configuring for device:vulkan://00000000-5500-0000-0000-000000000000
Using target triple -iree-vulkan-target-triple=rdna3-7900-windows from command line args
Saved vmfb in D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\euler_step_1_512_512_vulkan_fp16.vmfb.
Loading module D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\euler_step_1_512_512_vulkan_fp16.vmfb...
        Compiling Vulkan shaders. This may take a few minutes.
use_tuned? sharkify: False
_1_64_512_512_fp16_rockymixRecreated_v2
Diffusers' checkpoint will be identified here :  D:/Stable/stable-diffusion-shark/apps/stable_diffusion/web/models/diffusers/rockymixRecreated_v2
Loading diffusers' pipeline from original stable diffusion checkpoint
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["id2label"]` will be overriden.
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["bos_token_id"]` will be overriden.
`text_config_dict` is provided which will be used to initialize `CLIPTextConfig`. The value `text_config["eos_token_id"]` will be overriden.
D:\Stable\stable-diffusion-shark\shark.venv\Lib\site-packages\transformers\models\clip\feature_extraction_clip.py:28: FutureWarning: The class CLIPFeatureExtractor is deprecated and will be removed in version 5 of Transformers. Please use CLIPImageProcessor instead.
  warnings.warn(
Loading complete
Diffusers' checkpoint will be identified here :  D:/Stable/stable-diffusion-shark/apps/stable_diffusion/web/models/vae/diffusers/BerrysMix.vae
Loading diffusers' pipeline from original stable diffusion checkpoint
Processing standalone Vae checkpoint
mat1 and mat2 shapes cannot be multiplied (128x1024 and 768x320)
Retrying with a different base model configuration
D:\Stable\stable-diffusion-shark\shark.venv\Lib\site-packages\torch\fx\node.py:263: UserWarning: Trying to prepend a node to itself. This behavior has no effect on the graph.
  warnings.warn("Trying to prepend a node to itself. This behavior has no effect on the graph.")
saving unet_1_64_512_512_fp16_rockymixRecreated_v2_vulkan_torch_linalg.mlir to .
No vmfb found. Compiling and saving to D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\unet_1_64_512_512_fp16_rockymixRecreated_v2_vulkan.vmfb
Configuring for device:vulkan://00000000-5500-0000-0000-000000000000
Using target triple -iree-vulkan-target-triple=rdna3-7900-windows from command line args
Saved vmfb in D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\unet_1_64_512_512_fp16_rockymixRecreated_v2_vulkan.vmfb.
Loading module D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\unet_1_64_512_512_fp16_rockymixRecreated_v2_vulkan.vmfb...
        Compiling Vulkan shaders. This may take a few minutes.
::: Detailed report (took longer than 2.5s):
  +1.0027885437011719ms: get_iree_runtime_config
  +1.0027885437011719ms: mmap D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\unet_1_64_512_512_fp16_rockymixRecreated_v2_vulkan.vmfb
  +1.0027885437011719ms: ireert.SystemContext created
  +11395.0035572052ms: module initialized
saving clip_1_64_512_512_fp16_rockymixRecreated_v2_vulkan_torch_linalg.mlir to .
No vmfb found. Compiling and saving to D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\clip_1_64_512_512_fp16_rockymixRecreated_v2_vulkan.vmfb
Configuring for device:vulkan://00000000-5500-0000-0000-000000000000
Using target triple -iree-vulkan-target-triple=rdna3-7900-windows from command line args
Saved vmfb in D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\clip_1_64_512_512_fp16_rockymixRecreated_v2_vulkan.vmfb.
Loading module D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\clip_1_64_512_512_fp16_rockymixRecreated_v2_vulkan.vmfb...
        Compiling Vulkan shaders. This may take a few minutes.
saving unet512_1_64_512_512_fp16_rockymixRecreated_v2_vulkan_torch_linalg.mlir to .
No vmfb found. Compiling and saving to D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\unet512_1_64_512_512_fp16_rockymixRecreated_v2_vulkan.vmfb
Configuring for device:vulkan://00000000-5500-0000-0000-000000000000
Using target triple -iree-vulkan-target-triple=rdna3-7900-windows from command line args
Saved vmfb in D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\unet512_1_64_512_512_fp16_rockymixRecreated_v2_vulkan.vmfb.
Loading module D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\unet512_1_64_512_512_fp16_rockymixRecreated_v2_vulkan.vmfb...
        Compiling Vulkan shaders. This may take a few minutes.
::: Detailed report (took longer than 2.5s):
  +0.0ms: get_iree_runtime_config
  +1.0077953338623047ms: mmap D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\unet512_1_64_512_512_fp16_rockymixRecreated_v2_vulkan.vmfb
  +1.0077953338623047ms: ireert.SystemContext created
  +3854.9232482910156ms: module initialized
MLIR Textual PassPipeline Parser:1:11: error: 'iree-flow-detach-elementwise-from-named-ops' does not refer to a registered pass or pass pipeline
func.func(iree-flow-detach-elementwise-from-named-ops,iree-flow-convert-1x1-filter-conv2d-to-matmul,iree-preprocessing-convert-conv2d-to-img2col,iree-preprocessing-pad-linalg-ops{pad-size=32})
          ^
ERROR: mismatched parenthesis in pass pipeline `builtin.module(func.func(iree-flow-detach-elementwise-from-named-ops,iree-flow-convert-1x1-filter-conv2d-to-matmul,iree-preprocessing-convert-conv2d-to-img2col,iree-preprocessing-pad-linalg-ops{pad-size=32}))`
MLIR Textual PassPipeline Parser:1:11: error: 'iree-flow-detach-elementwise-from-named-ops' does not refer to a registered pass or pass pipeline
func.func(iree-flow-detach-elementwise-from-named-ops,iree-flow-convert-1x1-filter-conv2d-to-matmul,iree-preprocessing-convert-conv2d-to-img2col,iree-preprocessing-pad-linalg-ops{pad-size=32})
          ^
ERROR: mismatched parenthesis in pass pipeline `builtin.module(func.func(iree-flow-detach-elementwise-from-named-ops,iree-flow-convert-1x1-filter-conv2d-to-matmul,iree-preprocessing-convert-conv2d-to-img2col,iree-preprocessing-pad-linalg-ops{pad-size=32}))`
30it [00:09,  3.28it/s]
saving vae_1_64_512_512_fp16_rockymixRecreated_v2BerrysMix.vae_vulkan_torch_linalg.mlir to .
No vmfb found. Compiling and saving to D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\vae_1_64_512_512_fp16_rockymixRecreated_v2BerrysMix.vae_vulkan.vmfb
Configuring for device:vulkan://00000000-5500-0000-0000-000000000000
Using target triple -iree-vulkan-target-triple=rdna3-7900-windows from command line args
Saved vmfb in D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\vae_1_64_512_512_fp16_rockymixRecreated_v2BerrysMix.vae_vulkan.vmfb.
Loading module D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\vae_1_64_512_512_fp16_rockymixRecreated_v2BerrysMix.vae_vulkan.vmfb...
        Compiling Vulkan shaders. This may take a few minutes.
::: Detailed report (took longer than 2.5s):
  +0.0ms: get_iree_runtime_config
  +0.0ms: mmap D:\Stable\stable-diffusion-shark\apps\stable_diffusion\web\vae_1_64_512_512_fp16_rockymixRecreated_v2BerrysMix.vae_vulkan.vmfb
  +0.0ms: ireert.SystemContext created
  +9483.099699020386ms: module initialized
MLIR Textual PassPipeline Parser:1:11: error: 'iree-flow-detach-elementwise-from-named-ops' does not refer to a registered pass or pass pipeline
func.func(iree-flow-detach-elementwise-from-named-ops,iree-flow-convert-1x1-filter-conv2d-to-matmul,iree-preprocessing-convert-conv2d-to-img2col,iree-preprocessing-pad-linalg-ops{pad-size=32},iree-linalg-ext-convert-conv2d-to-winograd)
          ^
ERROR: mismatched parenthesis in pass pipeline `builtin.module(func.func(iree-flow-detach-elementwise-from-named-ops,iree-flow-convert-1x1-filter-conv2d-to-matmul,iree-preprocessing-convert-conv2d-to-img2col,iree-preprocessing-pad-linalg-ops{pad-size=32},iree-linalg-ext-convert-conv2d-to-winograd))`
30it [00:09,  3.27it/s]
30it [00:09,  3.25it/s]
30it [00:09,  3.23it/s]
30it [00:09,  3.33it/s]
30it [00:09,  3.30it/s]
30it [00:08,  3.34it/s]
30it [00:08,  3.37it/s]
30it [00:08,  3.38it/s]
```

Software / Hardware Information
* Windows 11 22H2
* RX 7900 XTX
* Adrenalin 23.9.3
* Vulkan 2.0.283
* [SHARK v4797bb8](https://github.com/nod-ai/SHARK/commit/4797bb89f5c5dbc7e40168e7dcb3c5cfa0e88235)

**Comment by monorimet** [comment]
Also seeing this issue.
```
Loading module C:\Users\ean\SHARK\unet_1_64_512_512_fp16_tuned_stable-diffusion-2-1-base_vulkan.vmfb...
        Compiling Vulkan shaders. This may take a few minutes.
::: Detailed report (took longer than 2.5s):
  +1.0018348693847656ms: get_iree_runtime_config
  +1.0018348693847656ms: mmap C:\Users\ean\SHARK\unet_1_64_512_512_fp16_tuned_stable-diffusion-2-1-base_vulkan.vmfb
  +1.0018348693847656ms: ireert.SystemContext created
  +10027.31704711914ms: module initialized
MLIR Textual PassPipeline Parser:1:11: error: 'iree-flow-detach-elementwise-from-named-ops' does not refer to a registered pass or pass pipeline
func.func(iree-flow-detach-elementwise-from-named-ops,iree-flow-convert-1x1-filter-conv2d-to-matmul,iree-preprocessing-convert-conv2d-to-img2col,iree-preprocessing-pad-linalg-ops{pad-size=32},iree-linalg-ext-convert-conv2d-to-winograd)
          ^
ERROR: mismatched parenthesis in pass pipeline `builtin.module(func.func(iree-flow-detach-elementwise-from-named-ops,iree-flow-convert-1x1-filter-conv2d-to-matmul,iree-preprocessing-convert-conv2d-to-img2col,iree-preprocessing-pad-linalg-ops{pad-size=32},iree-linalg-ext-convert-conv2d-to-winograd))`
50it [00:12,  4.15it/s]
```

Similar setup, 7900xtx, win11, adrenalin 23.20.11.01
seen on latest exe (996)

Something is gutting performance here, I usually see >20it/s

I can find out what happened to the flag here but would be (pleasantly) surprised if that fixes everything...

**Comment by one-lithe-rune** [comment]
Also got this and the performance regression.

**Comment by one-lithe-rune** [comment]
[993](https://github.com/nod-ai/SHARK/releases/tag/20231017.993) is the first nightly I see this issue on.

**Comment by one-lithe-rune** [comment]
I think this commit in SRT (presumably taken from upstream) might be the culprit https://github.com/nod-ai/SRT/commit/b7fd66860adf5a2fcd6b61c7cbbbe64f33474cd7

Commit message mentions both `Convert1X1FilterConv2DToMatmul`, `DetachElementwiseFromNamedOps`. Commenting those out removes the error and gets most of the performance back.

**Comment by njsharpe** [comment]
Checking 825b36c, will report back with issues or close if good.

**Comment by njsharpe** [comment]
@one-lithe-rune
New, seemingly related issue off fresh install (cleared caches & regenerated venv) off this commit.

```
MLIR Textual PassPipeline Parser:1:61: error: 'iree-global-opt-detach-convert-1x1-filter-conv2d-to-matmul' does not refer to a registered pass or pass pipeline
func.func(iree-global-opt-detach-elementwise-from-named-ops,iree-global-opt-detach-convert-1x1-filter-conv2d-to-matmul,iree-preprocessing-convert-conv2d-to-img2col,iree-preprocessing-pad-linalg-ops{pad-size=32})
                                                            ^
ERROR: mismatched parenthesis in pass pipeline `builtin.module(func.func(iree-global-opt-detach-elementwise-from-named-ops,iree-global-opt-detach-convert-1x1-filter-conv2d-to-matmul,iree-preprocessing-convert-conv2d-to-img2col,iree-preprocessing-pad-linalg-ops{pad-size=32}))`
MLIR Textual PassPipeline Parser:1:61: error: 'iree-global-opt-detach-convert-1x1-filter-conv2d-to-matmul' does not refer to a registered pass or pass pipeline
func.func(iree-global-opt-detach-elementwise-from-named-ops,iree-global-opt-detach-convert-1x1-filter-conv2d-to-matmul,iree-preprocessing-convert-conv2d-to-img2col,iree-preprocessing-pad-linalg-ops{pad-size=32})
                                                            ^
ERROR: mismatched parenthesis in pass pipeline `builtin.module(func.func(iree-global-opt-detach-elementwise-from-named-ops,iree-global-opt-detach-convert-1x1-filter-conv2d-to-matmul,iree-preprocessing-convert-conv2d-to-img2col,iree-preprocessing-pad-linalg-ops{pad-size=32}))`
```

Error occurs on Unet512 only, VAE error is gone.
Performance has not increased or decreased.

**Comment by njsharpe** [comment]
Resolved with 0361db4
